packages/common/src/hooks/useFilterBy/useFilterBy.tsx

```diff
@@ -11,39 +11,37 @@ type FilterRule = {
     | FilterByConditionByType['date']]?: unknown;
 };
 
-export type FilterBy<T> = Partial<Record<Partial<keyof T>, FilterRule>>;
+export type FilterBy = Record<string, FilterRule | null>;
```

Comment:
removing the `ObjectWithStringKeys` causes a whole bunch of issues everywhere.. This was taking a generic and then using a key to filter things by. 

I had some type issues with this, then I realised that we shouldn't be trying to restrict filtering to only keys on the object, otherwise things will be annoying. For example on related entity fields. 

So, issue to make #1 Make filtering somehow type safe..?  We can use the GQL types, maybe, so that type could be the generic here. But in general this filtering hook I don't really like and is quite difficult to use..

Reply:
I'm not seeing any use case where you'd filter by a field that isn't in the resulting data.. but there could well be something. in which case, making it type safe is an interesting challenge

Reply:
For example, the `InvoiceNode` - the total of the invoice becomes `invoice.pricing.totalAfterTax` - so if you wanted to do a filter for all invoices greater than X - or sort. Or filter by a transaction category which will also probably be nested 🤔